code/modules/projectiles/projectile/energy.dm

```diff
@@ -130,24 +130,22 @@
 	icon = 'icons/obj/apiary_bees_etc.dmi'
 	icon_state = "beegun"
 	check_armour = "bio"
-	damage = 0
+	damage = 5
 	damage_type = BRUTE
 	pass_flags = PASSTABLE | PASSGRILLE
 	embed = 0
 	weaken = 0
 
 /obj/item/projectile/energy/bee/on_impact(var/atom/A)
 	playsound(src.loc, pick('sound/effects/Buzz1.ogg','sound/effects/Buzz2.ogg'), 70, 1)
+	var/turf/T = get_turf(A)
 	for(var/i=1, i<=8, i++)
-		var/mob/living/simple_animal/bee/x = new /mob/living/simple_animal/bee
-		x.loc = A
-		x.feral = 25
-		if(ismob(A))
-			x.target_mob = A
-		if(prob(50))
-			for(var/j = 1, j <= rand(1, 3), j++)
-				step(x, pick(NORTH,SOUTH,EAST,WEST))
-
+		var/atom/movable/x = new /mob/living/simple_animal/bee/beegun //hackmaster pro, butt fuck it
+		x.loc = T
+		if(istype(T, /turf/simulated/wall) || istype(T, /turf/simulated/shuttle/wall))

```

Comment:
This is a worthless operation.

T is only picked once. Ergo, if it's a wall, window, or door, you run this loop 8 times, and delete the 8 groups of bees you've spawned. Put an if check before initiating the loop:

```
if (!istype(T, /turf/simulated/wall) && !istype(T, /turf/simulated/shuttle/wall) && !istype(A, /obj/structure/window) && !istype(A, /obj/machinery/door))
	for (var/i = 0, i < 7, i++)
		var/atom/movable/x = new /mob/living/simple_animal/bee/beegun
		x.forceMove(T)

..()
```